Add jsonschema format checker installation that does not install GPL-…

PhillSimonds opened the pull request:
Fixes #133

glennmatthews commented on CHANGELOG.md:
```diff
@@ -2,6 +2,10 @@
 
 ## v1.1.4 - 2022-06-16
 
+### Adds
```
This probably needs to be the start of a `v1.1.5` section right? Rather than falling under `v1.1.4`?

PhillSimonds replied:
I think we haven't formally released 1.1.4 yet, main is on 1.1.3, so was thinking both could be part of the `v1.1.4` release.

glennmatthews replied:
Right you are. I was misled by the presence of a release date for 1.1.4 in the changelog. :-)

PhillSimonds replied:
Ack. We normally include dates in the CHANGELOG.md correct? It was that there was a previous date we updated that misled?

glennmatthews replied:
Right, normally we only add a date for a given release when we actually release it :)

glennmatthews commented on poetry.lock:
```diff
@@ -686,6 +686,14 @@ python-versions = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*"
+python-versions = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*"
+
 [[package]]
 name = "rfc3987"
```
Something seems a bit off to me here - shouldn't this one be removed after the change in dependencies, since it's one of the GPL packages? Might try doing a `poetry remove jsonschema` and `poetry add jsonschema --extras format-nongpl` and see if that changes things at all.

PhillSimonds replied:
Hmmm. I ran those commands to remove jsonschema from the poetry env and re-install it with the format-nongpl. The command to remove did remove the rfc3986-validator package

```bash
(schema-enforcer-tr67_61X-py3.9) ➜  schema-enforcer git:(issue-133) poetry remove jsonschema
Updating dependencies
Resolving dependencies... (0.3s)

Writing lock file

Package operations: 0 installs, 0 updates, 9 removals

  • Removing arrow (1.2.2)
  • Removing fqdn (1.5.1)
  • Removing isoduration (20.11.0)
  • Removing jsonschema (4.7.1)
  • Removing pyrsistent (0.18.1)
  • Removing python-dateutil (2.8.2)
  • Removing rfc3986-validator (0.1.1)
  • Removing uri-template (1.2.0)
  • Removing webcolors (1.12)
```

The command to add with non-gpl added it back in.

```bash
(schema-enforcer-tr67_61X-py3.9) ➜  schema-enforcer git:(issue-133) ✗ poetry add jsonschema --extras format-nongpl                                                     
Using version ^4.7.1 for jsonschema

Updating dependencies
Resolving dependencies... (0.5s)

Writing lock file

Package operations: 9 installs, 0 updates, 0 removals

  • Installing python-dateutil (2.8.2)
  • Installing arrow (1.2.2)
  • Installing fqdn (1.5.1)
  • Installing isoduration (20.11.0)
  • Installing pyrsistent (0.18.1)
  • Installing rfc3986-validator (0.1.1)
  • Installing uri-template (1.2.0)
  • Installing webcolors (1.12)
  • Installing jsonschema (4.7.1)
```

PhillSimonds commented:
@glennmatthews it looks like we're failing unit tests on the [`incorrect_iri_format` check](https://github.com/networktocode/schema-enforcer/runs/7291284755?check_suite_focus=true). Per the [jsonschema python documentation](https://github.com/python-jsonschema/jsonschema/blob/main/docs/validate.rst), I believe that check requires the rfc3987 module, which is GPL licensed, so I think we may be out of luck?

glennmatthews commented:
Yeah, looks like we'll need to make a decision here. Some ideas:

- Remove the IRI check from our tests and add documentation that if you need to have that particular format check in your environment, it's up to you to install `rfc3987` yourself?
- Same, but also add a `gpl` extra to this package so that users can do `pip install schema-enforcer[gpl]` as a convenient shorthand for including `rfc3987` and `strict-rfc3339`?

PhillSimonds commented:
> Yeah, looks like we'll need to make a decision here. Some ideas:
> 
>     * Remove the IRI check from our tests and add documentation that if you need to have that particular format check in your environment, it's up to you to install `rfc3987` yourself?
> 
>     * Same, but also add a `gpl` extra to this package so that users can do `pip install schema-enforcer[gpl]` as a convenient shorthand for including `rfc3987` and `strict-rfc3339`?

Awesome. I believe we'd only need to have the users install `rfc3987` there. I've updated the README.md to include instructions, and removed the `iri` check. I don't think we'll need `strict-rfc3339` as per the [jsonschema documentation](https://github.com/python-jsonschema/jsonschema/blob/main/docs/validate.rst) `rfc3987` is the transitive dependency that defines those checks.

glennmatthews commented on CHANGELOG.md:
```diff
@@ -1,11 +1,19 @@
 
+### Removes
+
+- Native support for `iri` and `iri-reference` format checkers. This was removed because the format is implemented per the `rfc3987` jsonschema transitive dependency which is licensed under GPL.
```
Perhaps:

```suggestion
- Automatic support for `iri` and `iri-reference` format checkers. This was removed because these format checkers require the `rfc3987` library, which is licensed under GPL. If you require these checkers, you can manually install `rfc3987` or install this package as `jsonschema[rfc3987]`.
```
